---
title: "ClusterProfiler Analysis"
output: html_notebook
---


The goals for this analysis are to investigate how the gene expression changes based on the mouse model, genotype within the mouse model, and age.
Questions to answer:
-What genes are deferentially expressed due to Mouse Model Alone?
-How does aging affect gene expression within each mouse model?
-How does the genotype within each mouse model affect gene expression at different time points?

### Import experimental design matrix.

```{r}
library(DESeq2)
library(pheatmap)
library(dplyr)
library(dendextend)
```

```{r}
design_matrix<-read.table('/Users/tasnimtabassum/Documents/Transcriptomics_SP22/Experimental_Design_TG.csv',sep=',',header=TRUE)

head(design_matrix)
```

```{r}
rownames(design_matrix)<-design_matrix$Sample
design_matrix$Sample<-NULL

design_matrix
```

```{r}
counts_matrix<-read.table("/Users/tasnimtabassum/Documents/Transcriptomics_SP22/GitHub/Transcriptomics-Final-Project-/Count_Tables/allcounts.csv",sep=',',header=TRUE)

counts_matrix
```


Because the numbers after the dot in the ensembl IDs represent versions of genes in certain annotations, we can remove these to more easily conduct our differential gene expression analysis. 
```{r}
counts_matrix$V1<-gsub("\\..*","",counts_matrix$V1)

counts_matrix
```

```{r}
# remove the "V1" from col 1
rownames(counts_matrix)<-counts_matrix$V1
counts_matrix$V1<-NULL

head(counts_matrix)

counts_matrix<-counts_matrix[,order(colnames(counts_matrix))]
counts_matrix
```

```{r}
design_matrix<-design_matrix[order(rownames(design_matrix)),]

design_matrix

```

```{r}
dds <- DESeqDataSetFromMatrix(countData = counts_matrix,
                              colData = design_matrix,
                              design = ~ Genotype + Age+ Model)
dds

keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```


```{r}
dds <- DESeq(dds)
```

```{r}
normalizedcounts.matrix <- counts(dds,normalized=T)

res_1 <- results(dds, contrast = c("Genotype","rtg4510","WT_TG"))

res1_ordered <- res_1[order(res_1$padj),] 
head(res1_ordered)
```

Install Mouse annotation library:


```{r}
library(biomaRt) #For conversion of transcript IDs to gene ID
library(annotables) #to retrieve grcm38 annotation for mouse genome
library(org.Mm.eg.db) #Mouse genome annotation
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub) 
library(ensembldb)
library(tidyverse)
library(ggnewscale)
```

```{r}
# mouse genome load
grcm38

# check that ensgene in our df is prsent in the mouse genome df

idx <- grcm38$ensgene %in% rownames(res1_ordered)
head(idx)

# df with all the ids that are in our df from the mouse genome df
ids <- grcm38[idx, ]

head(ids)

# remove duplicates

non_duplicates <- which(duplicated(ids$ensgene) == FALSE)
ids <- ids[non_duplicates, ]
```

```{r}

#nrow(res1_ordered)

#rownames(res1_ordered)

# entrezID contains only the IDs that are also in our df

ensgeneID= grcm38[grcm38$ensgene %in% rownames(res1_ordered), ]

# check nrow entrezID
head(ensgeneID)

# create a vector of only the entrezIDs
ensgene_ID_vector = c(ensgeneID[[1]])

head(ensgene_ID_vector)

# create new df that contains only the ensgeneIDs, lfc and padj
res2= data.frame(log2foldchange= subset(res1_ordered$log2FoldChange, grcm38$ensgene %in% rownames(res1_ordered)))
padj = subset(res1_ordered$padj, grcm38$ensgene %in% rownames(res1_ordered))
res2 = cbind(padj, res2)
res2 = cbind(ensgene_ID_vector, res2)
gene_symbols = subset(grcm38$symbol, grcm38$ensgene %in% rownames(res1_ordered))
# omit all "na" values
res2 = cbind(gene_symbols, res2)
res2 = na.omit(res2)

head(res2)
```

```{r}
## Significant genes is a vector of fold changes where the names are ENTREZ gene IDs. The background set is a vector of all the genes represented on the platform.

# bg entrez contains all the ensgene

allOEgenes = as.character(res2$ensgene_ID_vector)

# sig res entrez contains all the entrezIDs that have padj <0.05
head(res2)

sigOE <- subset(res2, padj< 0.5)

head(sigOE)
# vector of only lfc values
sigOE_genes = as.character(sigOE$ensgene_ID_vector)

head(sigOE_genes)

```


```{r}
## Run GO enrichment analysis 
ego <- enrichGO(gene = sigOE_genes, 
                universe = allOEgenes,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.05, 
                readable = TRUE)
                
## Output results from GO analysis to a table
cluster_summary <- data.frame(ego)

head(cluster_summary)

head(ego)
#write.csv(cluster_summary, "results/clusterProfiler_Mov10oe.csv")

```


```{r fig.width=15,fig.height=20}
## Dotplot 


dotplot(ego, showCategory=10)
```





```{r fig.width=15,fig.height=20}
## To color genes by log2 fold changes, we need to extract the log2 fold changes from our results table creating a named vector
OE_foldchanges <- sigOE$log2foldchange

names(OE_foldchanges) <- sigOE$gene_symbols

## Cnetplot details the genes associated with one or more terms - by default gives the top 1 significant term (by padj)
cnetplot(ego, 
         categorySize="pvalue", 
         showCategory = 1, 
         foldChange=OE_foldchanges, 
         vertex.label.font=6)
         
## If some of the high fold changes are getting drowned out due to a large range, you could set a maximum fold change value
OE_foldchanges <- ifelse(OE_foldchanges > 1, 1, OE_foldchanges)
OE_foldchanges <- ifelse(OE_foldchanges < -1, -1, OE_foldchanges)

cnetplot(ego, 
         categorySize="pvalue", 
         showCategory = 1, 
         foldChange=OE_foldchanges, 
         vertex.label.font=6)
```


```{r}
ggrepel.max.overlaps = Inf
p1 <- cnetplot(ego,
               categorySize="pvalue", 
               showCategory = 4,
               node_label="all",
               cex_label_gene = 0.5,
               cex_label_category=0.9,
               shadowtext='category',
               color_category='grey', 
               color_gene='green')+ggtitle('Enriched Genes when comparing rTg4510 and wild-type rTg4510 mice')

p1
```

