---
title: "DESeq2 Multifactor Analysis Final Project"
output: html_notebook
---

# Goals for DGE Analysis

The goals for this analysis are to investigate how the gene expression changes based on the mouse model, genotype within the mouse model, and age.
Questions to answer:
-What genes are differentially expressed due to Mouse Model Alone?
-How does aging affect gene expression within each mouse model?
-How does the genotype within each mouse model affect gene expression at different time points?

### Import experimental design matrix.

```{r}
library(DESeq2)
library(pheatmap)
library(dplyr)
library(dendextend)
```


```{r}
design_matrix<-read.table('/Users/danielaquijano/Documents/GitHub/Transcriptomics-Final-Project-/source_files/Experimental_Design.csv',sep=',',header=TRUE)
```


```{r}
rownames(design_matrix)<-design_matrix$Sample
design_matrix$Sample<-NULL
```
```{r}
design_matrix
```

Import matrix containing the counts for all samples. These counts represent the forward strand counts since this was a stranded library. 

```{r}

counts_matrix<-read.table("/Users/danielaquijano/Documents/GitHub/Transcriptomics-Final-Project-/Count_Tables/allcounts.csv",sep=',',header=TRUE)
```
```{r}
counts_matrix
```
```{r}
rownames(counts_matrix)<-counts_matrix$V1
counts_matrix$V1<-NULL
```

```{r}
counts_matrix<-counts_matrix[,order(colnames(counts_matrix))]
```
```{r}
counts_matrix
```
```{r}
design_matrix<-design_matrix[order(rownames(design_matrix)),]
```
```{r}
design_matrix
```


```{r}
dds <- DESeqDataSetFromMatrix(countData = counts_matrix,
                              colData = design_matrix,
                              design = ~ Genotype + Age+ Model)
dds
```
Remove genes with counts less than 10

```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

Run DESeq
```{r}
dds <- DESeq(dds)
```

```{r}
rawcounts.matrix <- counts(dds,normalized=F)
normalizedcounts.matrix <- counts(dds,normalized=T)
class(rawcounts.matrix)
```

```{r}
vst_dds <- vst(dds)
dists <- dist(t(assay(vst_dds)))

```
```{r}
dists
```

TO DO: *****Label samples
```{r}

dendogram<-plot(hclust(dists),cex=0.6)

```
```{r}
labels(dendogram)
```

Trying to rename labels....
```{r}
dend <- dists %>%  hclust %>% as.dendrogram
dend %>% set("labels", c(rep('rtg4510',26),rep('J20',42),rep('rtg4510',4))) %>% set("labels_cex", 0.5) %>% plot
```

```{r}
head(vst_dds)
```

*** Add title, make graph cute (size bigger) Capitalize group
```{r}
plotPCA(vst_dds,intgroup=c("Model","Genotype","Age"))
```
# Obtain results based on different contrasts

### What genes are differentially expressed due to Mouse Model Alone?
```{r}
res_1 <- results(dds, contrast = c("Model","rtg4510","J20"))
```
```{r}
res1_ordered <- res_1[order(res_1$pvalue),] 
head(res1_ordered,200)
```

### -How does aging affect gene expression within each mouse model?
```{r}
res_2 <- results(dds, contrast = c("Age","2","8"))
```
```{r}
?results
```

-How does the genotype within each mouse model affect gene expression at different time points?


#GO-Term Enrichment

Install Mouse annotation library:
```{r}
BiocManager::install("org.Mm.eg.db")
```

```{r}
install.packages("devtools")
devtools::install_github("stephenturner/annotables")
```

```{r}
library(biomaRt) #For conversion of transcript IDs to gene ID
library(annotables) #to retrieve grcm38 annotation for mouse genome
library(org.Mm.eg.db) #Mouse genome annotation
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub) 
library(ensembldb)
library(tidyverse)
```


```{r}
grcm38
```

Because we have transcript IDs, we want ensembl ids in order to be able to conduct GO Term enrichment we need ensembl ID

```{r}
listEnsembl()
```

```{r}
ensembl <- useMart("ensembl", dataset="mmusculus_gene_ensembl")
annot<-getBM(c("ensembl_gene_id","ensembl_transcript_id", "mgi_symbol", "chromosome_name", "strand", "start_position", "end_position","gene_biotype"), mart=ensembl)
#searchDatasets(mart = ensembl, pattern = "mouse")
```

```{r}
annot
```

```{r}
## Use mouse genome 
ego <- enrichGO(gene = , 
                universe = ,
                keyType = "ENSEMBL",
                OrgDb = org.Mm.eg.db, 
                ont = "ALL", 
                pAdjustMethod = "BH", 
                qvalueCutoff = 0.05, 
                readable = TRUE)
                
## Output results from GO analysis to a table
enriched_genes_res1 <- data.frame(ego)
```


