---
title: "DESeq2 Multifactor Analysis Final Project"
output: html_notebook
---

# Goals for DGE Analysis

The goals for this analysis are to investigate how the gene expression changes based on the mouse model, genotype within the mouse model, and age.
Questions to answer:
-What genes are differentially expressed due to Mouse Model Alone?
-How does aging affect gene expression within each mouse model?
-How does the genotype within each mouse model affect gene expression at different time points?

### Import experimental design matrix.

```{r}
library(DESeq2)
library(pheatmap)
```


```{r}
design_matrix<-read.table('/Users/danielaquijano/Documents/GitHub/Transcriptomics-Final-Project-/source_files/Experimental_Design.csv',sep=',',header=TRUE)
```

```{r}
rownames(design_matrix)<-design_matrix$Sample
```
```{r}
design_matrix$Sample<-NULL
```
```{r}
design_matrix
```

Import matrix containing the counts for all samples. These counts represent the forward strand counts since this was a stranded library. 

```{r}

counts_matrix<-read.table("/Users/danielaquijano/Documents/GitHub/Transcriptomics-Final-Project-/Count_Tables/allcounts.csv",sep=',',header=TRUE)
```
```{r}
counts_matrix
```
```{r}
rownames(counts_matrix)<-counts_matrix$V1
counts_matrix$V1<-NULL
```

```{r}
counts_matrix<-counts_matrix[,order(colnames(counts_matrix))]
```
```{r}
counts_matrix
```
```{r}
design_matrix<-design_matrix[order(rownames(design_matrix)),]
```
```{r}
design_matrix
```


```{r}
dds <- DESeqDataSetFromMatrix(countData = counts_matrix,
                              colData = design_matrix,
                              design = ~ Genotype + Age+ Model)
dds
```
Remove genes with counts less than 10

```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

Run DESeq
```{r}
dds <- DESeq(dds)
```

```{r}
rawcounts.matrix <- counts(dds,normalized=F)
normalizedcounts.matrix <- counts(dds,normalized=T)
class(rawcounts.matrix)
```

